src/main/java/org/terasology/launcher/gui/javafx/ApplicationController.java

```diff
@@ -504,6 +523,56 @@ private void initComboBoxes() {
         });
     }
 
+    private void selectJobBoxItemForJob(final ComboBox<PackageItem> jobBox, final String jobId) {
```

Comment:
Both `selectJobBoxItemForJob` and `selectBuildVersionBoxItemForVersion` have an identical structure, so only difference is the `predicate` which is used to filter the items of the combo box.

Thus, we can use another abstraction here to select ~a specific~ the first item of a combo box matching a given predicate:

```java
    private <T> void selectFirst(final ComboBox<T> comboBox, Predicate<T> predicate) {
        final T item = comboBox.getItems().stream()
                .filter(predicate)
                .findFirst()
                .orElse(comboBox.getItems().get(0));

        comboBox.getSelectionModel().select(item);
    }
```

I personally disliked the if-else branching on the null check in the end, so maybe we can get the first element of the combo box as a default item, and just select whatever falls out of the stream in the end...

Reply:
The implementation of this method then becomes the following:

```java
private void selectJobBoxItemForJob(final ComboBox<JobItem> jobBox, final String jobId) {
    selectItem(jobBox, jobItem ->
                    jobItem.versionItems.stream()
                            .anyMatch(vItem -> vItem.linkedPackageProperty.get().getId().equals(jobId)));
}```

This is a bit more complicated than the other one, so it's probably best to keep the method. Maybe we can find a better name here (more focus on semantics and less on technical details)?